core/vm/suave.go

```diff
@@ -20,30 +30,10 @@ type SuaveContext struct {
 }
 
 type SuaveExecutionBackend struct {
-	ConfidentialStoreEngine *suave.ConfidentialStoreEngine
-	MempoolBackend          suave.MempoolBackend
+	ConfidentialStoreEngine ConfidentialStore
 	ConfidentialEthBackend  suave.ConfidentialEthBackend
 }
 
-func (b *SuaveExecutionBackend) Start() error {
```

Comment:
After the changes, I think the `ConfidentialStore` is a first-class citizen of the system and the initialization/lifecycle should be done as part of the whole `Ethereum` protocol initialization (piggybacking the `stack.RegisterLifecycle` utility) rather than doing it in the `SuaveExecutionBackend` that right now is only a wrapper around the `Suave API` for the precompiles.

Reply:
I'd say that it's still the engine that should manage the store